### DISCUSSION TITLE
Support for Go Projects

### DISCUSSION THREAD

**Discussion by bradleypeabody** [tailwindlabs/tailwindcss#2126] [Ideas]
I'd like to find out if there is any feedback on the idea of making a library for use in Go (pure Go, no JS or npm, etc.)

I'm doing work on Vugu (https://github.com/vugu/vugu) building out various examples and Tailwind CSS integration would be really useful.

The idea would be something like:
* The library would provide an interface for reading CSS and processing Tailwind CSS `@` directives, specifically `@tailwind` and `@apply`.  Possibly `@import` should be supported too.
* It would be callable both as a command-line tool or also from Go code.
* It would support outputting CSS files or Go source code to enable embedding files in Go programs easily.  (I think both the use case of Go programs embedding CSS to be served at run time, or doing the transform on the fly when the CSS file is requested via HTTP should both be well supported.)
* So far I'm looking at using https://github.com/tdewolff/parse/tree/master/css for CSS parsing, it should be quite fast.
* Mechanism to provide an allow-list of which style rules to include in the output might be useful as well.  I believe there is some support in Vue for example to scan templates and see which classes are actually used - I'm not sure I would go this far, at least not initially, but certainly some mechanism so there is a workable solution for cases where people want to trim their output down in size.  Even if it's a bit clunky to use (such as a callback function that can be implemented manually to say if a style should be included or not), it's probably better than nothing.
* It would of course live in its own separate repository and be written in Go (golang).
* For now I'm planning on using some simple scripts to embed the necessary tailwind CSS into the project so it do its work without any JS dependencies.
* While it's true my motivation for doing this is for Vugu projects, I can't think of anything that is Vugu-specific.  This library should be able to be used for any Go program that needs to render CSS that uses Tailwind at runtime or embed such CSS as part of its source.

Is this something anyone from the main Tailwind project (or anyone else) would like to have input on?  Any thoughts, suggestions, or cautionary tales?

**Comment by bradleypeabody** [comment]
FWIW: https://github.com/gotailwindcss/tailwind

**Comment by ghost** [comment]
+1 - I'm migrating a Bootstrap site with a Go backend to Tailwind, and i'd rather not setup a separate nodejs application just for purging/  Right now i'm just planning on using the CDN option, but this really isn't ideal and I hope something better for on-the-fly (or even something that runs initially with your code to do the file purging/generation) is developed.  https://github.com/gotailwindcss/tailwind is a good start, but I haven't been able to get it working for my specific use-case.
